### DISCUSSION TITLE
Is it safe up upgrade Debian?

### DISCUSSION THREAD

**Discussion by heliophane** [netbox-community/netbox#13517] [Q&A] [answered]
Hello all,

I'm running netbox with the tar file on a debian 12 VM. Would it be safe to upgrade debian to 12 without breaking any dependencies or permissions? the server is a production server so I don't want to break anything. Has anyone had experience with this?

**Comment by markkuleinio** [comment, marked as answer]
I've been running different NetBox versions on Debian 9, 10 and 11 for several years, I've always kept Debian up-to-date (`unattended-upgrades` installs security updates, other updates are installed with the usual "apt update && apt upgrade" process), I've never had any problems. I don't see why Debian 12 (the current stable Debian release) would be any different. That's the baseline of operating a production system, you have to keep it up-to-date.

If you are talking about upgrading Debian 11 to 12, I'm sure you can do that, but I have always had something else to improve as well so I've just installed a new Debian release on a new VM and migrated the NetBox database and configurations appropriately.

**Reply by heliophane** [reply]
Sweet, thank you. Yeah I've had unattended upgrades going for a while, just wanted to know if upgrading from 11 to 12 was safe.

**Comment by candlerb** [comment]
After upgrading the base OS, you may find that the Python version has changed, which will invalidate your virtualenv. But that's easily fixed by re-running the `./upgrade.sh` script again inside `/opt/netbox`.
